backend/src/auth/data_model/UserModel.ts

```diff
@@ -0,0 +1,31 @@
+import { Email } from "../../../../domain_model/Email"
+import { Uid } from "../../../../domain_model/Uid"
+
+
+export class UserModel {
+    id: Uid;
+    email: Email;
+    hashedPassword: string;
+    salt: string;
+
+    constructor(data: Pick<UserModel, "id" | "email" | "hashedPassword" | "salt">) {
```

Comment:
I haven't used Pick before, I'm assuming that his requires that id, email, hashedPassword, and salt are given as separate parameters and then it's combined into the data parameter?

Reply:
My understanding is its a subset of fields from the initial type (in this case UserModel).  This was effectively a shortcut from defining an interface with the same fields as the UserModel, and the constructor will take anything (including an anonymous object) that has all the fields in the Pick